core/dbt/config/runtime.py

```diff
@@ -328,10 +328,19 @@ def load_dependencies(self) -> Mapping[str, 'RuntimeConfig']:
         if self.dependencies is None:
             all_projects = {self.project_name: self}
             internal_packages = get_include_paths(self.credentials.type)
-            project_paths = itertools.chain(
-                internal_packages,
-                self._get_project_directories()
-            )
+            # raise exception if fewer installed packages than in packages.yml
+            package_dirs = self._get_project_directories()
+            count_packages_specified = len(self.packages.get('packages', 0))
+            count_packages_installed = len(tuple(package_dirs))
```

Comment:
Ah, I think (at least some of) the failing tests are because `_get_project_directories()` returns a generator object, so we can only access it once. No sense setting it to a var...
```suggestion
            count_packages_specified = len(self.packages.packages)
            count_packages_installed = len(tuple(self._get_project_directories()))
```